plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -277,13 +266,50 @@ public JdbcTableHandle getTableHandle(ConnectorSession session, PreparedQuery pr
                 ImmutableList.of(),
                 Optional.empty(),
                 OptionalLong.empty(),
-                Optional.of(columns.build()),
+                Optional.of(columns),
                 // The query is opaque, so we don't know referenced tables
                 Optional.empty(),
                 0,
                 Optional.empty());
     }
 
+    @Override
+    public JdbcProcedureHandle getProcedureHandle(ConnectorSession session, ProcedureQuery procedureQuery)
+    {
+        try (Connection connection = connectionFactory.openConnection(session);
+                CallableStatement callableStatement = queryBuilder.callProcedure(this, session, connection, procedureQuery)) {
+            ResultSetMetaData metadata = callableStatement.getMetaData();
```

Comment:
Can you add a test where multiple ResultSets are returned? Maybe we should fail in such case?

Reply:
Added test for it.